syntax = "proto2";

package flang;

enum SymbolType {
  SYMBOL_VARIABLE = 1;
  SYMBOL_FUNCTION = 2;
  SYMBOL_CLASS = 3;
}

enum DataType {
  INT32 = 1;
  INT64 = 2;
  CHAR = 3;
  BOOL = 4;
  STRING = 5;
  FLOAT = 6;
  ARRAY = 7;
  CLASS = 8;
}

// message BasicInfo {
  // // The full name is qualifier.name or name if qualifer equals empty.
  // required string name = 1;
  // required string qualifier = 2;
  // required SymbolType type = 3;
// }

// message FunctionInfoProto{
  // repeated string parameters = 1;
// }

// message ClassInfo {
// }

// message VariableInfo {
  // enum DataType {
    // INT32 = 1;
    // INT64 = 2;
    // FLOAT = 3;
    // CHAR = 4;
    // STRING = 5;
    // PRIMITIVE_TYPE = 100;
    // CLASS = 101;
  // }
  // required DataType data_type = 1;
  // optional ClassInfo class_info = 2;
// }
